xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp

```diff
@@ -786,21 +803,32 @@ void CLinuxRendererGLES::LoadShaders(int field)
       // Try GLSL shaders if supported and user requested auto or GLSL.
       if (glCreateProgram)
       {
-        // create regular progressive scan shader
-        m_pYUVShader = new YUV2RGBProgressiveShader(false, m_iFlags, m_format);
+        // create regular scan shader
         CLog::Log(LOGNOTICE, "GL: Selecting Single Pass YUV 2 RGB shader");
 
-        if (m_pYUVShader && m_pYUVShader->CompileAndLink())
+        m_pYUVProgShader = new YUV2RGBProgressiveShader(false, m_iFlags, m_format);
+        m_pYUVBobShader = new YUV2RGBBobShader(false, m_iFlags, m_format);
+        if ((m_pYUVProgShader && m_pYUVProgShader->CompileAndLink())
+            && (m_pYUVBobShader && m_pYUVBobShader->CompileAndLink()))
         {
           m_renderMethod = RENDER_GLSL;
           UpdateVideoFilter();
           break;
         }
-        else if (m_pYUVShader)
+        else
         {
-          m_pYUVShader->Free();
-          delete m_pYUVShader;
-          m_pYUVShader = NULL;
+          if (m_pYUVProgShader)
+          {
+            m_pYUVProgShader->Free();
+            delete m_pYUVProgShader;
+            m_pYUVProgShader = NULL;
+          }
+          if (m_pYUVBobShader)
+          {
+            m_pYUVBobShader->Free();
+            delete m_pYUVBobShader;
+            m_pYUVBobShader = NULL;
+          }
```

Comment:
The whole file deserves a big refactoring, really...
Plenty of duplication between the various render methods.